aws/table_aws_wafv2_regex_pattern_set.go

```diff
@@ -0,0 +1,426 @@
+		region = "us-east-1"
+		scope = aws.String("CLOUDFRONT")
+	}
+	plugin.Logger(ctx).Trace("listAwsWafv2RegexPatternSets", "AWS_REGION", region)
+
+	// Create session
+	svc, err := WAFv2Service(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// List all Regex Pattern Sets
+	pagesLeft := true
+	params := &wafv2.ListRegexPatternSetsInput{
+		Scope: scope,
+	}
+	for pagesLeft {
+		response, err := svc.ListRegexPatternSets(params)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, regexPatternSets := range response.RegexPatternSets {
+			d.StreamListItem(ctx, regexPatternSets)
+		}
+
+		if response.NextMarker != nil {
+			pagesLeft = true
+			params.NextMarker = response.NextMarker
+		} else {
+			pagesLeft = false
+		}
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getAwsWafv2RegexPatternSet(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getAwsWafv2RegexPatternSet")
+
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+
+	var id, name, scope string
+	if h.Item != nil {
+		data := regexPatternSetData(h.Item)
+		id = data["ID"]
+		name = data["Name"]
+		locationType := strings.Split(strings.Split(string(data["Arn"]), ":")[5], "/")[0]
+
+		if locationType == "regional" {
+			scope = "REGIONAL"
+		} else {
+			scope = "CLOUDFRONT"
+		}
+	} else {
+		id = d.KeyColumnQuals["id"].GetStringValue()
+		name = d.KeyColumnQuals["name"].GetStringValue()
+		scope = d.KeyColumnQuals["scope"].GetStringValue()
+	}
+
+	/*
+	 * The region endpoint is same for both Global Regex Pattern Set and the Regional Regex Pattern Set created in us-east-1.
+	 * The following checks are required to remove duplicate resource entries due to above mentioned condition, when performing GET operation.
+	 * To work with CloudFront, you must specify the Region US East (N. Virginia) or us-east-1
+	 * For the Regional Regex Pattern Set, region value should not be 'global', as 'global' region is only used to get Global Regex Pattern Sets.
+	 * For any other region, region value will be same as working region.
+	 */
+	if scope == "REGIONAL" && region == "global" {
+		return nil, nil
+	}
+
+	if strings.ToLower(scope) == "cloudfront" && region != "global" {
+		return nil, nil
+	}
+
+	if region == "global" {
+		region = "us-east-1"
+	}
+
+	// Create Session
+	svc, err := WAFv2Service(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	params := &wafv2.GetRegexPatternSetInput{
+		Id:    aws.String(id),
+		Name:  aws.String(name),
+		Scope: aws.String(scope),
+	}
+
+	op, err := svc.GetRegexPatternSet(params)
+	if err != nil {
+		plugin.Logger(ctx).Debug("GetRegexPatternSet", "ERROR", err)
+		return nil, err
+	}
+
+	return op.RegexPatternSet, nil
+}
+
+func getAwsWafv2RegexPatternSetLockToken(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

Comment:
We can avoid using get api call twice rather we can use new version of `FromField` where we can pass more than one path in comma separated  form.

Reply:
updated